### ISSUE TITLE
ScrollablePositionedList: There seems to be a problem with the widget's state when back to the previous screen when if use listen.

### ISSUE THREAD

**Issue by oseiasmribeiro** [google/flutter.widgets#54]
I'm taking the scroll index to save the position in the database that the user stopped reading. For this I am using a Listen as shown in the image. Except that when returning to the previous screen it seems to give a bug related to the state of the ScrollablePositionedList.

![Captura de Tela 2020-01-24 às 09 20 22](https://user-images.githubusercontent.com/36827173/73068950-42c7fb80-3e8b-11ea-86c8-6c23139aafd4.png)

**Comment by tarobins** [comment]
what's the `+ 2` for in that example?

**Comment by oseiasmribeiro** [comment]
![Captura de Tela 2020-08-03 às 13 05 37](https://user-images.githubusercontent.com/36827173/89203091-5961b480-d58a-11ea-846a-1346f76a4add.png)
@tarobins Sorry I'm late. I use +2 for ScrollablePositionList to position beyond the header added with SliverPersistentHeader. But, strangely, this problem didn't occur to me after the recent updates.

**Comment by oseiasmribeiro** [comment]
@tarobins If the ScrollablePositionedList is inside the NestedScrollView I don't get the exception, but as I now removed it, it occurred to me again.

**Comment by oseiasmribeiro** [comment]
@tarobins I found that the problem was in the future response from Navigator.push.Then () when I returned to the previous screen. I was erroneously told to clear a list in the Bloc, this made Flutter go through the entire widget tree again until it reached the ScrollablePositionedList, and as the list was empty, the bad state error occurred. But the question remains, didn't the ScrollablePositionedList have to predict this? An example repository follows to analyze where the bad state error occurs when returning. If this is not an error feel free to close the problem. Thank you!
[https://github.com/oseiasmribeiro/scrollablepositionedlistbadstate.git](https://github.com/oseiasmribeiro/scrollablepositionedlistbadstate.git)

> ...
            Navigator.push(
              context,
              MaterialPageRoute(builder: (context) => ReadingMessage()),
            ).then(
              (isSearchBar){
                _bloc.listTextParagraph.clear(); // PROBLEM IS HERE
                _bloc.changeLongPressForShare(false);
                FocusScope.of(context).requestFocus(FocusNode());
              }
            );
    ...
